router/http_test.go

```diff
@@ -401,6 +409,65 @@ func (s *S) TestStickyHTTPRoute(c *C) {
 	}
 }
 
+func (s *S) TestStickyHTTPRouteWebsocket(c *C) {
+	wshandler := func(w http.ResponseWriter, req *http.Request) {
+		w.Write([]byte("ok " + strings.TrimPrefix(req.URL.Path, "/")))
+	}
+	srv1 := httptest.NewServer(http.HandlerFunc(wshandler))
+	srv2 := httptest.NewServer(http.HandlerFunc(wshandler))
+	defer srv1.Close()
+	defer srv2.Close()
+
+	l, discoverd := newHTTPListener(c)
+	defer l.Close()
+
+	addStickyHTTPRoute(c, l)
+
+	discoverdRegisterHTTP(c, l, srv1.Listener.Addr().String())
+	defer discoverd.UnregisterAll()
+
+	// triggers the TCP-mode conn upgrade path, but does not hit the websocket
+	// handler and so the upgrade is not successful.
+	assertGetCookieUpgradeClosed := func(c *C, url, host, expected string, cookie *http.Cookie) *http.Cookie {
```

Comment:
FYI @benburkert has reorganized this test case so that it's much clearer what's going on: https://github.com/bgentry/flynn/pull/1

Will bring that in after the last couple of tweaks.